### ISSUE TITLE
Performance is fairly good :)

### ISSUE THREAD

**Issue by scott-r-lindsey** [thecodingmachine/graphqlite-bundle#5]
Hi there, I'm seeing an overhead of about 120ms per query (vs. 40ms for overblog/GraphQLBundle) on my lightly loaded linux dev machine.

I would so much prefer to use this bundle -- the configuration is just so much better, but a difference of 80ms per query seems disqualifying.

Do you think this is something that can be improved, or this just an inherent trade-off?

Thanks for reading.

EDIT -- the original title for this issue was "Performance is not very good" but it seems that my methodology was not very good.

**Comment by moufmouf** [comment]
Hey Scott,

Yes, performance is a major concern right now for us.

There are a number of optimizations we have already applied but that you might not have noticed.

In order to be developer friendly, we are scanning all controller classes / type classes for annotations. We are obviously caching those, but we are still scanning the list of PHP files to see if one has been modified. This explains the 120 ms you see.

What you probably don't see is that there is a very short lived cache (2 seconds) in order to avoid overloading the server with scans of files. So we are scanning all PHP files once every 2 seconds. When you are developing, you don't notice the cache, and if your server is under high load, it will actually behave ok (you will hit the cache quite often) and response time will be <120ms.

That being said, we need to add a "production" mode that is using a long-lived cache instead of a short lived one.

Also, to be completely transparent, what GraphQLite lacks the most right now is the support for Deferred objects. We are also working on this feature (an issue is already open here: https://github.com/thecodingmachine/graphqlite/issues/14)

**Comment by scott-r-lindsey** [comment]
Thank you for explaining all of that.  

So what I did, is I re-did my test, but this time I edited graphqllite so that the $globTTL is a bit higher, and what I found is that for a for my trivial test on one entity, overblog is giving me a result in ~55ms, while graphqlite is coming in perhaps 10ms slower, which to me is a totally reasonable outcome, not at all disqualifying.

I'm not done though -- I'm going to do a test with a larger query on each platform and see how it goes.
